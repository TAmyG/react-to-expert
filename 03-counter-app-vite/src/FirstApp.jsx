import React from 'react'

export const FirstApp = ({title= "Hola a todos"}) => {
  return (
    <div>FirstApp: {props.title}</div>
  )
}
