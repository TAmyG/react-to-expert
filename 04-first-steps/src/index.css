body {
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

html,
body {
  background-color: #282c34;
  color: white;
  font-size: calc(10px + 2vmin);
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
  text-align: center;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}

button {
  font-size: 1.3rem;
  margin-right: 10px;
  padding: 10px;
}
