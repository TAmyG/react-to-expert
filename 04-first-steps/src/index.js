import React from 'react';
import ReactDOM from 'react-dom';
//import App from './App';
import reportWebVitals from './reportWebVitals';
import CounterApp from './CounterApp';
import PrimeraApp from './PrimeraApp';
import './index.css';

ReactDOM.render(
  <React.StrictMode>
    <CounterApp value={10} />
    {/*     <PrimeraApp saludo="holamundo" /> */}
  </React.StrictMode>,
  document.getElementById('root'),
);

// If you want to start measuring performance in your app, pass a function
// to log results (for example: reportWebVitals(console.log))
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
reportWebVitals();
