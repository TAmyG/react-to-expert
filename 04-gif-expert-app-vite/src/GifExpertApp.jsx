/**
 * 
 * After react v17 is not necesary the React import
 */

import { useState } from "react";
import { AddCategory } from "../components/AddCategory";
import { GifGrid } from "../components/GifGrid";
/**
 * 
 * 1. crear state
 * 2. agregar categoria. Investiguen como agregar un elemento
 * 3. Crear componente Add Category
 * 4. Enviar prop setCategories
 * 5. Eliminar button Agregar
 * 6. Tarea implmentar con onAddCategory mejor solucion ya que el componente debe emitir el valor
 * 7. Evitar duplicados
 * 8. Nuevo Componente GifGrid
 */
export const GifExpertApp = () => {
    //1.
    const [categories, setCategories] = useState(['One Punch']);

    /** 
    * 2.
    * No se puede hacer push directo a categories, ya que React trata de no mutar el state
    * Lo preferible es crear un nuevo state con setCategories
    */
    const onAddCategory = (newCategory) => {
        //1.
        //setCategories(prev => ['Naruto', ...prev]);

        //3.
        if(categories.includes(newCategory)) return;
        //2.
        setCategories(prev => [newCategory, ...prev]);

    }

    return (
        <>
            {/**Titulo */}
            <h1>GifExpertApp</h1>
            {/**Input */}
            {/* <AddCategory setCategories={setCategories} />
            <button onClick={onAddCategory} >Agregar</button> */}
            <AddCategory onNewCategory={onAddCategory} />
            <ol>
                {/* {categories.map(category => <li key={category} >{category}</li>)} */}
                {categories.map(category => <GifGrid key={category} category={category}/>)}
            </ol>

            {/**Listado de Gifs */}
            {/**Gif Item*/}

        </>
    )
}
