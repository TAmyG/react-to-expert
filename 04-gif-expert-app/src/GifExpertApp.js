import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({ defaultCategories = [] }) => {
    // Hooks
    //const [categories, setCategories] = useState(['One Punch']);
    const [categories, setCategories] = useState(defaultCategories);

    return (
        <div>
            <h2>GifExperApp</h2>
            <AddCategory setCategories={setCategories} />
            <hr />

            <ol>
                {categories.map((category) => (
                    <GifGrid key={category} category={category} />
                ))}
            </ol>
        </div>
    );
};
