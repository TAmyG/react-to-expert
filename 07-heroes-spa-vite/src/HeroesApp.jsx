import { AuthProvider } from "./auth"
import { AppRouter } from "./router/AppRouter"

/**
 * 
 * Parte final hacer authProvider
 */
export const HeroesApp = () => {
  return (
    <AuthProvider>
      <AppRouter />
    </AuthProvider>
  )
}
