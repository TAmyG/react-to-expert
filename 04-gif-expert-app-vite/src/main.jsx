import React from 'react'
import ReactDOM from 'react-dom/client'
import { GifExpertApp } from './GifExpertApp'

import './styles.css'; //estilos globales

/**
 * StricMode ayuda a identificar problemas o advertencias con nuestro codigo y se aplica solo en desarrollo
 */
ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GifExpertApp />
  </React.StrictMode>
);
