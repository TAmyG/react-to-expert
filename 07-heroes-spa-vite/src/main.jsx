import React from 'react';
import ReactDOM from 'react-dom/client';
import { BrowserRouter } from "react-router-dom";

import { HeroesApp } from './HeroesApp';
import './index.css';

/**
 * 1. Crear estructura de carpetas
 * 2. Crear AppRouter.jsx
 */
ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <BrowserRouter>
      <HeroesApp />
    </BrowserRouter>
  </React.StrictMode>
)
