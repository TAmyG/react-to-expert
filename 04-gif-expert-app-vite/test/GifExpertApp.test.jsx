import { screen, render, fireEvent } from '@testing-library/react';
import { GifExpertApp } from '../src/GifExpertApp';


describe('Pruebas en <GifExpertApp/>', () => {


    test('debe hacer match con el snapshot', () => {
        const { container } = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();
    });
})