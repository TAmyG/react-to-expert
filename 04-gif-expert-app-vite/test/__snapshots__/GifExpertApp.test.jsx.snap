// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp/> debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
  <ol>
    <h3>
      One Punch
    </h3>
    <h1>
      Loading...
    </h1>
    <div
      class="card-grid"
    />
  </ol>
</div>
`;
